pysb/simulator/stochkit.py

```diff
@@ -0,0 +1,305 @@
+
+    >>> from pysb.examples.robertson import model
+    >>> import numpy as np
+    >>> np.set_printoptions(precision=4)
+    >>> sim = StochKitSimulator(model, tspan=np.linspace(0, 10, 5))
+
+    Here we supply a "seed" to the random number generator for deterministic
+    results, but for most purposes it is recommended to leave this blank.
+    >>> simulation_result = sim.run(n_runs=2, seed=123456)
+
+    A_total trajectory for first run
+    >>> print(simulation_result.observables[0]['A_total']) \
+        #doctest: +ELLIPSIS
+    [ 1.  0.  0.  0.  0.  0.]
+
+    A_total trajectory for second run
+    >>> print(simulation_result.observables[1]['A_total']) \
+        #doctest: +ELLIPSIS
+    [ 1.  1.  1.  1.  0.  0.]
+
+    For further information on retrieving trajectories (species,
+    observables, expressions over time) from the ``simulation_result``
+    object returned by :func:`run`, see the examples under the
+    :class:`SimulationResult` class.
+    """
+    _supports = {'multi_initials': True, 'multi_param_values': True}
+
+    def __init__(self, model, tspan=None, initials=None,
+                 param_values=None, verbose=False, **kwargs):
+        super(StochKitSimulator, self).__init__(model,
+                                                tspan=tspan,
+                                                initials=initials,
+                                                param_values=param_values,
+                                                verbose=verbose,
+                                                **kwargs)
+        self.cleanup = kwargs.get('cleanup', True)
+        self._outdir = None
+        generate_equations(self._model,
+                           cleanup=self.cleanup,
+                           verbose=self.verbose)
+
+    def _run_stochkit(self, t=20, t_length=100, number_of_trajectories=1,
+                      seed=None, algorithm='ssa', method=None,
+                      num_processors=1, stats=False, epsilon=None,
+                      threshold=None):
+
+        extra_args = '-p {:d}'.format(num_processors)
+
+        # Random seed for stochastic simulation
+        if seed is not None:
+            extra_args += ' --seed {:d}'.format(seed)
+
+        # Keep all the trajectories by default
+        extra_args += ' --keep-trajectories'
+
+        # Number of trajectories
+        extra_args += ' --realizations {:d}'.format(number_of_trajectories)
+
+        # We generally don't need the extra stats
+        if not stats:
+            extra_args += ' --no-stats'
+
+        if method is not None:  # This only works for StochKit 2.1
+            extra_args += ' --method {}'.format(method)
+
+        if epsilon is not None:
+            extra_args += ' --epsilon {:f}'.format(epsilon)
+
+        if threshold is not None:
+            extra_args += ' --threshold {:d}'.format(threshold)
+
+        # Find binary for selected algorithm (SSA, Tau-leaping, ...)
+        if algorithm is None:
```

Comment:
If the default was `None` then this would be a good error message to have. However the default is 'ssa' so someone would have to explicitly pass `None` to fail this check, and there's a test against an explicit list of valid simulator names below anyway, so you can drop this.